ibm/resource_ibm_cis_healthcheck.go

```diff
@@ -2,331 +2,364 @@ package ibm
 			return err
 		}
+		log.Printf("Monitor update succesful : %s", *result.Result.ID)
 	}
-	return resourceCIShealthCheckRead(d, meta)
+
+	return resourceCISHealthCheckRead(d, meta)
 }
 
-func resourceCIShealthCheckDelete(d *schema.ResourceData, meta interface{}) error {
-	cisClient, err := meta.(ClientSession).CisAPI()
+func resourceCISHealthCheckDelete(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).CisGLBHealthCheckClientSession()
 	if err != nil {
 		return err
 	}
-	monitorId, cisId, err := convertTftoCisTwoVar(d.Id())
+
+	monitorID, crn, err := convertTftoCisTwoVar(d.Id())
 	if err != nil {
 		return err
 	}
-	var monitor *v1.Monitor
-	emptyMonitor := new(v1.Monitor)
-	monitor, err = cisClient.Monitors().GetMonitor(cisId, monitorId)
+	sess.Crn = core.StringPtr(crn)
+
+	opt := sess.NewDeleteLoadBalancerMonitorOptions(monitorID)
+
+	result, resp, err := sess.DeleteLoadBalancerMonitor(opt)
 	if err != nil {
-		if checkCisMonitorDeleted(d, meta, err, monitor) {
-			d.SetId("")
-			return nil
-		}
-		log.Printf("[WARN] Error getting zone during MonitorRead %v\n", err)
+		log.Printf("Error deleting global load balancer health check detail: %s", resp)
 		return err
 	}
-
-	monitorObj := *monitor
-	if !reflect.DeepEqual(emptyMonitor, monitorObj) {
-		err = cisClient.Monitors().DeleteMonitor(cisId, monitorId)
-		if err != nil {
-			log.Printf("[WARN] DeleteMonitor Failed %s\n", err)
-			return err
-		}
-	}
-
-	d.SetId("")
+	log.Printf("Monitor ID: %s", *result.Result.ID)
 	return nil
 }
 
-func checkCisMonitorDeleted(d *schema.ResourceData, meta interface{}, errCheck error, monitor *v1.Monitor) bool {
-	// Check if error is due to removal of Cis resource and hence all subresources
-	if strings.Contains(errCheck.Error(), "Object not found") ||
-		strings.Contains(errCheck.Error(), "status code: 404") ||
-		strings.Contains(errCheck.Error(), "Invalid zone identifier") { //code 400
-		log.Printf("[WARN] Removing resource from state because it's not found via the CIS API")
-		return true
+func resourceCISHealthCheckExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	sess, err := meta.(ClientSession).CisGLBHealthCheckClientSession()
+	if err != nil {
+		return false, err
 	}
-	_, cisId, _ := convertTftoCisTwoVar(d.Id())
-	exists, errNew := rcInstanceExists(cisId, "ibm_cis", meta)
-	if errNew != nil {
-		log.Printf("resourceCISmonitorRead - Failure validating service exists %s\n", errNew)
-		return false
+
+	monitorID, crn, err := convertTftoCisTwoVar(d.Id())
+	if err != nil {
+		return false, err
 	}
-	if !exists {
-		log.Printf("[WARN] Removing monitor from state because parent cis instance is in removed state")
-		return true
+	sess.Crn = core.StringPtr(crn)
+
+	opt := sess.NewGetLoadBalancerMonitorOptions(monitorID)
+
+	result, response, err := sess.GetLoadBalancerMonitor(opt)
+	if err != nil {
+		if response != nil &&
+			(response.StatusCode == 404 ||
+				response.StatusCode == 400) {
```

Comment:
400 is bad request it should return error

Reply:
removed